### PR TITLE
feat(foundryup): Support local installation in foundryup

### PR DESCRIPTION
**Pull request by dmfxyz**
## Motivation
This PR addresses issue #929 by adding the ability to install forge and cast from local repositories.

## Solution
Add a `--path` argument to the `foundryup` up script. The value of the argument should be a path to a local foundry repo. Then:

1. cd into the path
2. run cargo build
3. remove any existing `forge` or `cast` binaries that exist in the `$FOUNDRY_BIN_DIR`
4. symlink the locally installed binaries to `$FOUNDRY_BIN_DIR/{forge, cast}`

The symlinks are destroyed whenever a new remote or release version of Foundry is installed using `foundryup`

## Tests Run
At each step in each test I confirmed that the expected version of foundry was installed

1. fresh foundry install using --path, --repo, and release
2. release -> local -> remote
3. release -> remote -> local
4. remote -> release ->  local
5. remote -> local -> release
6. local -> remote -> release
7. local -> release -> remote

Todo
- [x] Update Help Message
- [x] Update README
- [ ] Integrate code review feedback

### CODE REVIEW

File: foundryup/install [hunk 1]
```diff
@@ -37,10 +38,42 @@ OPTIONS:
...
+  fi
+
+  if ! command -v cargo &> /dev/null ; then
+    # Error if cargo is not already installed.
+    echo "foundryup: cargo is not installed. Please install it first."
+    exit 1
+  fi
+
+  # Enter local repo and build
+  FOUNDRY_REPO="$PWD/$FOUNDRYUP_LOCAL_REPO"
```
Comment (onbjerg) [inline foundryup/install:61]: This will fail if you use an absolute path. I think just using `$FOUNDRYUP_LOCAL_REPO` should handle both cases?

**Review by onbjerg — CHANGES_REQUESTED** [review]
Other than that lgtm, will need to test it before merging :)

File: foundryup/install (hunk 1 above)
Comment (gakonst) [reply foundryup/install:61]: agreed - would prefer we use absolute paths, shell expansion should cover converting it otherwise

Comment (dmfxyz) [reply foundryup/install]: agreed! This was left over from some relative-path symlinking debugging. Pushed changes that following @onbjerg suggestion and confirmed it supports both absolute and relative paths. Also added some testing notes in the PR.

**Comment by dmfxyz** [comment]
@gakonst @onbjerg I think this is ready for final review and merging when you have a moment :)

**Review by gakonst — APPROVED** [review]
👌👌

**Pull request merged**